refactor: separate external/bundled completion directories

akinomyoga opened the pull request:
This implements https://github.com/scop/bash-completion/discussions/1329#discussioncomment-12163012.

The first commit b4dda6d12 contains purely renaming of files (plus necessary changes to `.pre-commit-config.yaml` to make `git commit` to pass). This is to make `git` correctly recognize the move of the files. See the commit message for details.
- In this PR, the name `-core` (instead of `-dist` initially suggested in [#1329](https://github.com/scop/bash-completion/discussions/1329#discussioncomment-12163012))  is adopted. Also, fallback completions are currently separated into `-fallback`. I think the naming of the directories needs to be discussed before merging.
  - The background for the choice of `core` is explained in [this reply](https://github.com/scop/bash-completion/discussions/1329#discussioncomment-14771602): `dist` seems to be used by both the core and third-party packages in the case of `texmf` and used to exclude the files installed by the users and administrators of a specific machine.
  - Currently fallback completions are separated into `completions-fallback`, but it was suggested in [a reply](https://github.com/scop/bash-completion/discussions/1329#discussioncomment-14766343) to not distinguish the primarily maintained completions and fallback/compat completions and mix the files in a directory `completions-core`. However, I would like to keep the distinction as argued in [a reply](https://github.com/scop/bash-completion/discussions/1329#discussioncomment-14771885).

The second commit 29355f8cf adjusts the file contents for the file renaming. The search order of `_comp_load` is also adjusted. New search ordering is described in the updated `README.md`.

- The search ordering may also be a topic to be discussed.
- To simplify the PR, the following substantial changes were separated into the next (planned) PR.
  - I plan to change the precedence of `completions/<cmd>` vs `completions/<cmd>.bash` in the next PR (so that `<cmd>.bash` will have the higher priority than `<cmd>`)
  - I plan to remove the support for `completion/_<cmd>` because `_<cmd>` has been reserved for internal uses of `bash-completion` and is not supposed to be used by external completions, as has been described in `README.md`.
  - The pkg-config variable `helpersdir` will be described in `README`.

The third commit 961b8be83 removes `test/fallback/completions`. See the commit message for the details.

scop reviewed (approved):
This makes sense to me, including the proposed naming. I have not read everything through, but I'm comfortable going with this if @yedayak agrees. Whether we go with this or not, let's not keep this cooking for too long as this stuff is very prone to conflicts.

yedayak commented on completions/Makefile.am:
```diff
@@ -1,1575 +1,22 @@
-install-data-hook: ss = $(SETUP_SYMLINKS) $(DESTDIR)$(bashcompdir)
+install-data-hook: ss = $(SETUP_SYMLINKS) $(DESTDIR)$(bashcompdir_fallback)
 install-data-hook: symlinks
 
```
We should probably still create the directory when installing, so external completions can be placed here easily
Maybe using something [like this](https://stackoverflow.com/a/8531548)?

Same for `helpers/` if users are expected to put their own stuff there.

yedayak commented on completions/Makefile.am:
We should probably still create the directory when installing, so external completions can be placed here easily
Maybe using something [like this](https://stackoverflow.com/a/8531548)?

Same for `helpers/` if users are expected to put their own stuff there.

akinomyoga replied:
I also added the instructions to create/remove the directories in the installation target (f8b9f781a0b3a2acadda7962e392bb30e4fa8f6f...946457e30a499262df48bb74f2f55b5e9298aff0).

akinomyoga commented:
CI started to fail with the committed change (#1525). It seems the maximum length of the commit subject became 50 (instead of the previous 70), which seems too short to me. Can we explicitly set the maximum length to 70, or is this change intentional?

```
f8b9f78: error Commit subject is too long, 66 exceeds the max length of 50
```

akinomyoga commented on test/Makefile.am:
```diff
@@ -1,8 +1,12 @@
 	fixtures/make/extra_makefile
+
+clean-local:
+	-rm -rf config/bash-completion config/bash-completion-fallback
```
Thank you. I added the directories to `.gitignore` (f8b9f781a0b3a2acadda7962e392bb30e4fa8f6f...946457e30a499262df48bb74f2f55b5e9298aff0).

akinomyoga commented on test/config/bashrc:
```diff
@@ -1,5 +1,15 @@
+    if [[ -L $dst || -e $dst ]]; then
+        rm -rf "$dst"
+    fi
+    mkdir -p "${dst%/*}"
```
OK, so `make distcheck` seems to require that the check does not modify anything in the source tree. One way is to create the directories in a temporary directory, but I'd like to think about a better way.

scop commented:
Re commit subject length, https://github.com/scop/bash-completion/pull/1532

yedayak reviewed (approved):
For a followup: some exclusions in pre-commit for completions-core are probably not needed anymore, since all of the files end with `.bash` now. Also some file filters could be simplified to just catch `*.bash` files.

akinomyoga commented:
> For a followup: some exclusions in pre-commit for completions-core are probably not needed anymore, since all of the files end with `.bash` now. Also some file filters could be simplified to just catch `*.bash` files.

Thank you. I updated: [`92d9a8d..6fb0bf9`](https://github.com/scop/bash-completion/compare/92d9a8d6b1262fdcb02d7859c555a5862b97f1e5..6fb0bf929e44fb258741fd117f3288e4ef864936). Actually, some of `completions-core` in `.pre-commit-config.yaml` should have been `completions-(core|fallback)`. I added `\.bash` and removed some `exclude`s.
